Source/Core/Core/PowerPC/Jit64/Jit.cpp

```diff
@@ -615,35 +615,40 @@ const u8* Jit64::DoJit(u32 em_address, PPCAnalyst::CodeBuffer *code_buf, JitBloc
 	js.skipInstructions = 0;
 	js.carryFlagSet = false;
 	js.carryFlagInverted = false;
-	js.assumeNoPairedQuantize = false;
-
-	// If the block only uses one GQR and the GQR is zero at compile time, make a guess that the block
-	// never uses quantized loads/stores. Many paired-heavy games use largely float loads and stores,
-	// which are significantly faster when inlined (especially in MMU mode, where this lets them use
-	// fastmem).
-	// Insert a check that the GQR is still zero at the start of the block in case our guess turns out
-	// wrong.
-	// TODO: support any other constant GQR value, not merely zero/unquantized: we can optimize quantized
-	// loadstores too, it'd just be more code.
-	if (code_block.m_gqr_used.Count() == 1 && js.pairedQuantizeAddresses.find(js.blockStart) == js.pairedQuantizeAddresses.end())
+	js.constantGqr.clear();
+
+	// Assume that GQR values don't change often at runtime. Many paired-heavy games use largely float loads and stores,
+	// which are significantly faster when inlined (especially in MMU mode, where this lets them use fastmem).
+	if (js.pairedQuantizeAddresses.find(js.blockStart) == js.pairedQuantizeAddresses.end())
 	{
-		int gqr = *code_block.m_gqr_used.begin();
-		if (!code_block.m_gqr_modified[gqr] && !GQR(gqr))
+		// Ensure that the GQRs used are constant
+		for (int gqr : code_block.m_gqr_used)
```

Comment:
Have you considered using `std::find_if` and simply doing a normal `if` instead of the `goto`?